File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int MaxDepth(TreeNode root, int height = 0) {
        if(root == null)
            return 0;
        return (1+Math.Max(MaxDepth(root.left, height+1), MaxDepth(root.right, height+1)));
    }
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    int maxHeight = 0;
    void height(TreeNode* root, int currHeight) {
        if(root == NULL)
            return;
        if(currHeight > maxHeight)
            maxHeight = currHeight;
        height(root->left, currHeight+1);
        height(root->right, currHeight+1);
    }
public:
    int maxDepth(TreeNode* root) {
        if(root == NULL)
            return 0;
        height(root, 1);
        return maxHeight;
    }
};